src/Workspaces/Remote/ServiceHub/Services/LanguageServer.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.NavigateTo;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json;
+using Roslyn.Utilities;
+using StreamJsonRpc;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    // we need per language server for now since ILanguageClient
+    // doesn't allow multiple content types to be associated with
+    // one language server
+    internal class CSharpLanguageServer : LanguageServer
```

Comment:
>CSharpLanguageServer [](start = 19, length = 20)

I'd be useful to add a comment on all types that are listed in .servicehub.service.json file that these are instantiated by service hub and that GenerateServiceHubConfigurationFiles.targets needs to be updated if the type is renamed.